src/main/java/com/spotify/dns/DirectChangeNotifier.java

```diff
@@ -16,11 +16,10 @@
 
 package com.spotify.dns;
 
-import com.google.common.base.Supplier;
+import static java.util.Objects.requireNonNull;
 
 import java.util.Set;
-
-import static com.google.common.base.Preconditions.checkNotNull;
+import java.util.function.Supplier;
```

Comment:
This seems like an API breaking change? So then this should be a major version change?

Reply:
This is a package private class. I've kept the public parts of the package backwards compatible. Also relied to `japicmp` to catch incompatible changes.